core/chains/evm/client/pool.go

```diff
@@ -195,36 +196,41 @@ func (p *Pool) ChainID() *big.Int {
 	return p.chainID
 }
 
-func (p *Pool) getRoundRobin() Node {
-	nodes := p.liveNodes()
-	nNodes := len(nodes)
-	if nNodes == 0 {
-		p.logger.Critical("No live RPC nodes available")
-		return &erroringNode{errMsg: fmt.Sprintf("no live nodes available for chain %s", p.chainID.String())}
+func (p *Pool) getBestNode() Node {
+	highestHeadNumber := int64(-1)
+	var node Node
+	for _, n := range p.nodes {
+		latestReceivedBlockNumber := n.LatestReceivedBlockNumber()
+		if n.State() == NodeStateAlive && n.LatestReceivedBlockNumber() > highestHeadNumber {
```

Comment:
Careful, `n.LatestReceivedBlockNumber` may have changed between calls.

Reply:
This is a bug, will fix. It should be `latestReceivedBlockNumber` var.

Reply:
Fixed